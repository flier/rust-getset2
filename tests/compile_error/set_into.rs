use getset2::{Getter, Setter};

#[derive(Getter, Setter)]
pub struct Struct {
    #[get(pub, copy)]
    #[set(pub, into)]
    field: f32,
}

fn main() {
    let mut s = Struct { field: 3.14 };

    assert_eq!(s.set_field(123i32).field(), 123.0);
}
