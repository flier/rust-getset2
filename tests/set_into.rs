use getset2::{Getter, Setter};

#[derive(Default, Getter, Setter)]
pub struct Foo {
    #[get(str)]
    #[set(into)]
    into_field: String,
}

#[test]
fn set_into() {
    let mut foo = Foo::default();

    assert_eq!(foo.set_into_field("bar").into_field(), "bar");
}
