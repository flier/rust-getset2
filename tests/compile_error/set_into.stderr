error[E0277]: the trait bound `f32: From<i32>` is not satisfied
  --> tests/compile_error/set_into.rs:13:28
   |
13 |     assert_eq!(s.set_field(123i32).field(), 123.0);
   |                  --------- ^^^^^^ the trait `From<i32>` is not implemented for `f32`
   |                  |
   |                  required by a bound introduced by this call
   |
   = help: the following other types implement trait `From<T>`:
             <f32 as From<bool>>
             <f32 as From<i8>>
             <f32 as From<i16>>
             <f32 as From<u8>>
             <f32 as From<u16>>
   = note: required for `i32` to implement `Into<f32>`
note: required by a bound in `Struct::set_field`
  --> tests/compile_error/set_into.rs:5:5
   |
3  | #[derive(Getter, Setter)]
   |                  ------ required by a bound in this associated function
4  | pub struct Struct {
5  |     #[get(pub, copy)]
   |     ^ required by this bound in `Struct::set_field`
